alert-common/src/main/java/com/synopsys/integration/alert/common/action/UploadEndpointManager.java

```diff
@@ -90,83 +88,74 @@ public void unRegisterTarget(String targetKey) throws AlertException {
         uploadTargets.remove(targetKey);
     }
 
-    public ResponseEntity<String> performUpload(String targetKey, Resource fileResource) {
+    public void performUpload(String targetKey, Resource fileResource) {
         if (!containsTarget(targetKey)) {
-            return new ResponseEntity(NO_UPLOAD_FUNCTIONALITY_REGISTERED, HttpStatus.NOT_IMPLEMENTED);
+            throwNotImplementedException();
         }
```

Comment:
Can we use the entire if statement in the throwNotImplementedException method? We perform this exact check multiple times. We'd just change the name of the throw method to throwNotImplementedExceptionIfTargetDoesntExist or something not as long